libs/sdk-core/src/persist/swap.rs

```diff
@@ -245,6 +245,10 @@ impl SqliteStorage {
         self.select_single_swap("swaps.bitcoin_address = ?1", [address])
     }
 
+    pub(crate) fn get_swap_info_by_bolt11(&self, bolt11: &str) -> Result<Option<SwapInfo>> {
```

Comment:
The existing `get_swap_info_by_hash` could probably be used here instead.

If we later add a `swaps_info` INDEX on `payment_hash`, it would benefit this and other callers too.

Reply:
I agree we should use hash instead of bolt11. The main reason is because the bolt11 that exists on the swap is the one that the payer pays and the bolt11 that persisted with the payment is the one that the node settled. 
In case where a channel is opened as part of the swap these can be different but same hash.

Reply:
@ok300 @roeierez Sure, makes sense. I'm working on using a single query, but I'm having issues implementing the join due to the hash being stored as a hex string on the payments table and as a blob on the swaps table. Any suggestions on how to address this?

I see the following options:
* change the payment_hash column type to string on the swaps table (requiring encoding and decoding where necessary in the current places the table it's currently used)
* persist a serialized `SwapInfo` object in `payments_external_info` in a similar fashion to how other objects are persisted at the moment (this should be fine as long as we don't care about changes to the swap information after we've redeemed the swap)

Reply:
Good catch.

We could also look into the [hex](https://www.sqlite.org/lang_corefunc.html#hex) and [unhex](https://www.sqlite.org/lang_corefunc.html#unhex) core SQLITE functions. Either of them could probably already be used in the join (e.g. `WHERE hex(payment_hash_blob) == payment_hash_hex`).

If we change the DB schema to converge on a common representation of hashes (for example hex), we could also use one of these functions to migrate data "on-the-fly" in a new migration step.

What do you think?

Btw if we change the schema, it might be better to split this in 2 steps:

- first, DB migration to common hash format, which also migrates existing data
- second, add this PR on top, which can then join without having to convert blob to hex or viceversa

I can do the migration PR if you'd like.

Reply:
> We could also look into the [hex](https://www.sqlite.org/lang_corefunc.html#hex) and [unhex](https://www.sqlite.org/lang_corefunc.html#unhex) core SQLITE functions. Either of them could probably already be used in the join (e.g. WHERE hex(payment_hash_blob) == payment_hash_hex).

Oh, I didn't know of these functions and didn't think of searching for them. Thanks!

> Alternatively, if we change the DB schema to converge on a common representation of hashes (for example hex), we could also use one of these functions to migrate data "on-the-fly" in a new migration step.

Hm, probably makes sense to switch to a common representation. Can you elaborate on why it's better to do the migration PR first? I think it would be fine to have a query using `hex()` and remove later.

I've just updated the PR to use the hex function within the payment fetching queries.

Reply:
You're right, its also fine to use `hex()` for now and remove it later with the migration.